app/models/pipeline_run_stage.rb

```diff
@@ -29,6 +29,13 @@ class PipelineRunStage < ApplicationRecord
   DAG_NAME_POSTPROCESS = "postprocess".freeze
   DAG_NAME_EXPERIMENTAL = "experimental".freeze
 
+  STEP_NUMBER_TO_DAG_JSON_NAMES = {
```

Comment:
I think we should avoid redefining the ordering here because the dag model is meant to (take steps to) decouple the stage orderings from an exact sequence. So PipelineRunStage.step_number is meant to handle the ordering. So as an alternative you could have a mapping of the stage names to the "dag JSON" names like "Host Filtering" => "host_filter".

Reply:
I would actually just merge this with a structure that defines ALL properties (dag to STAGES.
Would then change `create_run_stages` to use it.
Didnt we agree on this for the currently blocked PR?

Reply:
I guess we missed the extra map that is done on the `create_run_stages`. We should consolidate that one too.

Reply:
Do you mean a mapping from stage number => stage name?

Reply:
I meant defining a unique structure like:
```
STAGE_INFO = {
	1 => {
		name: "Host Filtering".freeze,
		dag_name: "host_filtering".freeze,
		job_command_func: "host_filtering_command".freeze
	},
	...
```

And then always using this same structure to feed all places where you need to access this info.

Reply:
I added in a map like above, but also left the original constant definitions above since some parts of the code reference the name or dag_name directly (for example, the dag_json templates or the mapping of descriptions in `pipeline_runs_helper`), and going along with Jonathan's comment, I don't want to directly relate those (especially the dag_json template) with ordering.

Reply:
Jonathan meant that you should always use the same order that is injected into the DB. And that order should be defined in a unique place.
At this point it is defined it in two different places that you need to keep in sync.